### PR TITLE
Added parent child resolvers

### PR DESCRIPTION
**Pull request by likhitha-surapaneni**
Ticket: [ENSVAR-6050](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-6050)

### CODE REVIEW

File: common/file_model/variant_allele.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    def __init__(self, allele_index: str, alt: str, variant:dict) -> None:
+        
+        self.name = f"{variant.chromosome}:{variant.position}:{variant.ref}:{alt}"
+        self.variant = variant
+        self.allele_index = allele_index
+        self.alt = alt
+        self.type = "VariantAllele"
+        self.allele_sequence = alt
+        self.reference_sequence = variant.ref
+        self.info_map = self.traverse_csq_info()
```
Comment (nakib103) [inline common/file_model/variant_allele.py:18]: Should we be creating the full map all at once here?
Currently it fails to retrieve any allele field even if one of those that rely on this map fails, for example `Polyphen` in yeast.

File: common/file_model/variant_allele.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        return self.info_map[min_alt]["prediction_results"] if min_alt in self.info_map else []
+    
+    
+            
+    def traverse_csq_info(self) -> Mapping:
+        """
+        This function is to traverse the CSQ record and extract columns
+        corresponding to Consequence, SIFT, PolyPhen, CADD
+        """
+        allele_index = self.variant.get_info_key_index("Allele")
```
Comment (nakib103) [inline common/file_model/variant_allele.py:49]: Some of this field is not guaranteed to be in the data file, for example - `Conservation` or `CADD` or `Feature`. It depends on the species and type of variants (e.g - `intergenic`). We should not break a api if a field or value is missing, rather return null or empty value depending on the schema.

**Review by nakib103 — CHANGES_REQUESTED** [review]

File: common/file_model/variant_allele.py (hunk 1 above)
Comment (likhitha-surapaneni) [reply common/file_model/variant_allele.py:18]: Working on this in [ENSVAR-6050](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-6050)

File: common/file_model/variant_allele.py (hunk 2 above)
Comment (likhitha-surapaneni) [reply common/file_model/variant_allele.py:49]: Thanks you @nakib103 , working on this in [ENSVAR-6050](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-6050)

**Review by nakib103 — APPROVED** [review]
Great, thanks Likhitha!

**Pull request merged**